### PR TITLE
[Gekidou] refactor clean notifications

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
This PR refactors the way we clean push notifications that have been received.

Fixes the logout Android crash when attempting to clean notifications

Try and keep the app icon badge in sync with remaining mentions across servers.

Note: the sync will get totally out of sync when a user mark a post as unread [ticket here](https://mattermost.atlassian.net/browse/MM-46467)

```release-note
NONE
```

### CODE REVIEW

File: app/init/push_notifications.ts [hunk 1]
```diff
@@ -52,25 +52,9 @@ class PushNotifications {
...
-    };
-
-    cancelChannelsNotifications = async (channelIds: string[]) => {
-        const notifications = await NativeNotifications.getDeliveredNotifications();
-        for (const channelId of channelIds) {
-            this.cancelNotificationsForChannel(notifications, channelId);
-        }
-    };
-
     cancelNotificationsForChannel = (notifications: NotificationWithChannel[], channelId: string, rootId?: string, isCRTEnabled?: boolean) => {
```
Comment (anurag6713) [inline app/init/push_notifications.ts:55]: Shall we remove this function as it's not called anymore?

Comment (enahum) [reply app/init/push_notifications.ts:55]: I thought I removed it.. thanks

**Review by anurag6713 — APPROVED** [review]

**Comment by enahum** [comment]
@anurag6713 please re-review, I've added a few changes and split them in commits to ease the review...

**Review by anurag6713 — APPROVED** [review]
Great work! 👏 🫡

**Pull request merged**